### PR TITLE
FOUNDATIONS: Retrieve Language By Id

### LINKED ISSUE #28: FOUNDATIONS: Retrieve Language By Id

### PR DESCRIPTION
**Pull request by iCloudBMX**
Closes #28

### CODE REVIEW

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                this.languageService.RetrieveLanguageByIdAsync(invalidLanguageId);
+
+            // then
+            await Assert.ThrowsAsync<LanguageValidationException>(() =>
+                retrieveLanguageByIdTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLanguageValidationException))),
+                    Times.Once);
```
Comment (hassanhabib) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs:44]: Push this line forward by one tab

Comment (hassanhabib) [reply Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs:44]: Do the same for the rest.

File: Sofee.Core.Api/Services/Foundations/Languages/LanguageService.cs [hunk 2]
```diff
@@ -34,5 +35,18 @@ public partial class LanguageService : ILanguageService
 
             return await this.storageBroker.InsertLanguageAsync(language);
         });
+
+        public ValueTask<Language> RetrieveLanguageByIdAsync(Guid languageId) =>
+            TryCatch(async () =>
```
[the remark this answers is not in the document]
Comment (hassanhabib) [reply Sofee.Core.Api/Services/Foundations/Languages/LanguageService.cs:40]: @iCloudBMX it's not:

image

[the remark this answers is not in the document]
Comment (iCloudBMX) [reply Sofee.Core.Api/Services/Foundations/Languages/LanguageService.cs]: done

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls(); 
+        }
+
+        [Fact]
+        public async Task ShouldThrowNotFoundExceptionOnRetrieveByIdIfLanguageIsNotFoundLogItAsync()
+        {
+            // given
+            Guid randomLanguageId = Guid.NewGuid();
+            Language? noLanguage = null;
```
Comment (ShriHumrudha) [inline Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs:60]: Why ?

File: Sofee.Core.Api/Models/Languages/Exceptions/NotFoundLanguageException.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (ShriHumrudha) [inline Sofee.Core.Api/Models/Languages/Exceptions/NotFoundLanguageException.cs:1]: Copyrights

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs (hunk 3 above)
Comment (iCloudBMX) [reply Sofee.Core.Api.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Validation.RetrieveById.cs]: done

File: Sofee.Core.Api/Models/Languages/Exceptions/NotFoundLanguageException.cs (hunk 4 above)
Comment (iCloudBMX) [reply Sofee.Core.Api/Models/Languages/Exceptions/NotFoundLanguageException.cs]: done

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**